torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -477,17 +477,20 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupMPI::gather(
   }
   checkSingleTensor(inputTensors);
 
-  if (outputTensors.size() != 1) {
+  if (inputTensors.size() != 1) {
```

Comment:
Why do you need this check? checkSingleTensor has done this already in line 478

Reply:
Removing.